stdlib/list.gr

```diff
@@ -842,6 +846,48 @@ provide let sub = (start, length, list) => {
   take(length, drop(start, list))
 }
 
+// List.join helpers
+@unsafe
+let rec iterSize = (sepSize: WasmI32, size: WasmI32, lst: List<String>) => {
+  use WasmI32.{ (+) }
+  use DataStructures.{ stringSize }
+  match (lst) {
+    [] => size,
+    [hd, ...tl] => {
```

Comment:
If you add a case where there's only one list element, you can avoid having to pass a negative value for `size`. The case with just one element adds only the size of that element and the case with more than one adds the size plus the separator size and recurses.